db/version_set.cc

```diff
@@ -4903,10 +4903,27 @@ bool VersionStorageInfo::RangeMightExistAfterSortedRun(
 }
 
 Env::WriteLifeTimeHint VersionStorageInfo::CalculateSSTWriteHint(
-    int level) const {
-  if (compaction_style_ != kCompactionStyleLevel) {
+    int level, CompactionStyleSet compaction_style_set) const {
+  if (compaction_style_ != kCompactionStyleLevel &&
```

Comment:
This is awkward code. If the style is in the set, we should go to a switch statement for handling.